STYLEGUIDE.md

```diff
@@ -0,0 +1,175 @@
+# Style guide and design principles
+
+This guide provides a set of design principles and guidelines for the `yoneda`
+project. Our style and design principles take heavy inspiration from
+[`agda-unimath`](https://github.com/UniMath/agda-unimath).
+
+## The structure of code
+
+We enforce strict formatting rules. This formatting allows the type of the
+defined term to be easily readable, and aids in understanding the structure of
+the definition.
+
+The general format of a definition is as follows:
+
+```rzk
+#def concat
+  ( p : x = y)
+  ( q : y = z)
+  : (x = z)
+  := idJ (A , y , \ z' q' -> (x = z') , p , z , q)
+```
+
+- We start with the name, and place every assumption on a new line.
+
+- The conclusion of the definition is placed on its own line, which starts with
+  a colon (`:`).
+
+- Then, on the next line, the walrus separator (`:=`) is placed, and after it
+  follows the actual construction of the definition. If the construction does
+  not fit on a single line, we immediately insert a new line after the walrus
+  separator and indent the code an extra level (2 spaces).
+
+(Currently just taken from agda-unimath and adapted to Rzk) In Rzk, every
+construction is structured like a tree, where each operation can be seen as a
+branching point. We use indentation levels and parentheses to highlight this
+structure, which makes the code feel more organized and understandable. For
+example, when a definition part extends beyond a line, we introduce line breaks
+at the earliest branching point, clearly displaying the tree structure of the
+definition. This allows the reader to follow the branches of the tree, and to
+visually grasp the scope of each operation and argument. Consider the following
+example about Segal types:
+
+```rzk
+#def is-segal-is-local-horn-inclusion
+  ( A : U)
+  ( is-local-horn-inclusion-A : is-local-horn-inclusion A)
+  : isSegal A
+  :=
+    \ x y z f g ->
+    projection-equiv-contractible-fibers
+      ( < {t : 2 * 2 | Λ t} -> A >)
```

Comment:
Angle brackets (`<`, `>`) are unnecessary starting from `rzk-v0.2.0`.
I suggest to make them deprecated entirely and drop in one of the future versions.